### PR TITLE
feat(streampetr): fix bugs and refactor code

### PR DESCRIPTION
**Pull request by SamratThapa120**
This pull request introduces several improvements and refactorings to the StreamPETR 3D detection pipeline, focusing on dataset handling, configuration, and deployment scripts. The most significant changes enhance camera handling, dataset filtering, reproducibility, and configuration clarity.

**Dataset and camera handling improvements:**
* The dataset pipeline in `dataset.py` now supports explicit camera ordering, anchor camera selection, and camera shuffling, improving flexibility and reproducibility in multi-camera scenarios. The code asserts the anchor camera is present and adds logic for shuffling cameras during training. [[1]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL61-R73) [[2]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL82-L95) [[3]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL275-R266)
* The dataset filtering logic is refactored to ensure only valid entries with existing image paths and anchor cameras are included, and the sorting is now based on anchor camera timestamps. This change improves robustness against missing or invalid data.
* Sequence start times are tracked per sample, and timestamps are now normalized relative to the sequence start, which supports more accurate temporal modeling. [[1]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eR133-R163) [[2]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL263-R246) [[3]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL275-R266)

**Configuration changes for training and reproducibility:**
* Camera order is explicitly set in configuration files, removing ambiguity and ensuring consistent camera input order. [[1]](diffhunk://#diff-51e90b74ca05cc679a366f06f81bd3809f6351b142f26a13991eebb12859adefL24-R25) [[2]](diffhunk://#diff-7950911ed7bc1b146bed317536be0bee9483e161637fd61e6228078787b5f195L24-L26)
* Model weights are now loaded from remote URLs for baseline configurations, facilitating reproducible experiments and easier setup. [[1]](diffhunk://#diff-51e90b74ca05cc679a366f06f81bd3809f6351b142f26a13991eebb12859adefL427-R430) [[2]](diffhunk://#diff-7950911ed7bc1b146bed317536be0bee9483e161637fd61e6228078787b5f195L426-R428)
* A `randomness` configuration block is added to enforce deterministic training runs and fixed seeds, improving reproducibility. [[1]](diffhunk://#diff-51e90b74ca05cc679a366f06f81bd3809f6351b142f26a13991eebb12859adefL427-R430) [[2]](diffhunk://#diff-7950911ed7bc1b146bed317536be0bee9483e161637fd61e6228078787b5f195L426-R428)

**Training pipeline and hyperparameter tuning:**
* New and updated configuration files introduce batch size, data root, optimizer, and dataloader changes, as well as new sampler logic for streaming datasets. This supports more efficient training and easier hyperparameter tuning. [[1]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L6-R9) [[2]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014R18-R27) [[3]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L40-R55) [[4]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L53-L54) [[5]](diffhunk://#diff-9c553d6d3b8d63f27f99020c94bea50814c4400667668fca74ca36c56e5e48cbR1-R66)

**Deployment and ONNX export fixes:**
* The deployment script now uses the correct number of cameras from the configuration and fixes data types for position embedding export, improving compatibility with downstream tools. [[1]](diffhunk://#diff-7d3b58be57f17677250349179a68db65f6901c9a5e85ce104f383b51fb7b40afL90-R90) [[2]](diffhunk://#diff-7d3b58be57f17677250349179a68db65f6901c9a5e85ce104f383b51fb7b40afL156-R156)

**Miscellaneous fixes:**
* The class order in nuScenes configuration files is updated for consistency. [[1]](diffhunk://#diff-49275b07e3b756e200b2094f620d565d0205cfae1c6267804474973e779dfd90L37-R44) [[2]](diffhunk://#diff-2e62b2e373ec03197358ba2895d5117b78f515d37d3805cc794bcbce2ae4a167L37-R44)
* Assertion on sample tokens is removed from evaluation to allow more flexible evaluation splits.

---

**Dataset and camera handling:**
- Added explicit camera order, anchor camera selection, and shuffling logic to `StreamPETR` dataset pipeline, with assertions and improved sequence/timestamp handling. [[1]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL61-R73) [[2]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL82-L95) [[3]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eR133-R163) [[4]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL263-R246) [[5]](diffhunk://#diff-a464bc9ae53938f1e79cec3bcabd29b538b9873921b38079771d952c7b886e6eL275-R266)
- Refactored dataset filtering and sorting to ensure only valid samples with anchor camera images are used, improving robustness.

**Configuration and reproducibility:**
- Set camera order explicitly and loaded pretrained weights from remote URLs in default configs; added deterministic randomness settings for reproducibility. [[1]](diffhunk://#diff-51e90b74ca05cc679a366f06f81bd3809f6351b142f26a13991eebb12859adefL24-R25) [[2]](diffhunk://#diff-7950911ed7bc1b146bed317536be0bee9483e161637fd61e6228078787b5f195L24-L26) [[3]](diffhunk://#diff-51e90b74ca05cc679a366f06f81bd3809f6351b142f26a13991eebb12859adefL427-R430) [[4]](diffhunk://#diff-7950911ed7bc1b146bed317536be0bee9483e161637fd61e6228078787b5f195L426-R428)

**Training pipeline and hyperparameters:**
- Updated and added configuration files for various dataset splits, batch sizes, optimizers, and dataloaders, including new sampler logic for streaming. [[1]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L6-R9) [[2]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014R18-R27) [[3]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L40-R55) [[4]](diffhunk://#diff-1eb1e5cff98ec7fad1d4f87d121499e3d9dd52b068b247a446dc5d20c511b014L53-L54) [[5]](diffhunk://#diff-9c553d6d3b8d63f27f99020c94bea50814c4400667668fca74ca36c56e5e48cbR1-R66)

**Deployment and ONNX export:**
- Fixed deployment script to use correct camera count and data types for ONNX export. [[1]](diffhunk://#diff-7d3b58be57f17677250349179a68db65f6901c9a5e85ce104f383b51fb7b40afL90-R90) [[2]](diffhunk://#diff-7d3b58be57f17677250349179a68db65f6901c9a5e85ce104f383b51fb7b40afL156-R156)

**Miscellaneous:**
- Updated class order in nuScenes configs for consistency; removed strict assertion in evaluation for more flexible splits. [[1]](diffhunk://#diff-49275b07e3b756e200b2094f620d565d0205cfae1c6267804474973e779dfd90L37-R44) [[2]](diffhunk://#diff-2e62b2e373ec03197358ba2895d5117b78f515d37d3805cc794bcbce2ae4a167L37-R44) [[3]](diffhunk://#diff-343b9f379295c8d06414807b4e69adbfb93c07915fc3e33dbf0c95e4f8d045a5L97-L99)

### CODE REVIEW

**Comment by SamratThapa120** [comment]
@KSeangTan The performance did not change for test set. it was more or less same in terms of mAP.  This is likely due to test set having consistent spacing between frames. It is just my assumption that this kind of augmentation will help make the model robust during inference.

File: projects/StreamPETR/stream_petr/datasets/pipelines/dataset.py [hunk 1]
```diff
@@ -260,7 +262,7 @@ def get_annot_info(self, index):
             next_idx=info.get("next", None),
             scene_token=info["scene_token"],
             frame_idx=info["token"],
-            timestamp=info["timestamp"],
+            timestamp=info["images"][self.anchor_camera]["timestamp"],
```
[the remark this answers is not in the document]
Comment (SamratThapa120) [reply projects/StreamPETR/stream_petr/datasets/pipelines/dataset.py:239]: I added checks to see if all camera images exists during data validation, so it should check for anchor camera too.

File: projects/StreamPETR/stream_petr/datasets/samplers/group_streaming_sampler.py [hunk 2]
```diff
@@ -68,7 +70,10 @@ def _compute_indices(self, epoch: int):
             # Divide selected_groups into self.batch_size groups, drop the last if not divisible
             batch_groups = [[] for _ in range(self.batch_size)]
             for i, group in enumerate(selected_groups):
-                batch_groups[i % self.batch_size].extend(group)
+                # Randomly drop some elements from the group based on the random_drop_probability, to simulate camera drop
+                batch_groups[i % self.batch_size].extend(
+                    [x for x in group if np.random.rand() > self.random_drop_probability]
```
[the remark this answers is not in the document]
Comment (SamratThapa120) [reply projects/StreamPETR/stream_petr/datasets/samplers/group_streaming_sampler.py:75]: FIXED

File: projects/StreamPETR/stream_petr/models/detectors/petr3d.py
[the remark this answers is not in the document]
Comment (SamratThapa120) [reply projects/StreamPETR/stream_petr/models/detectors/petr3d.py:1]: FIXED

**Review by yamsam — APPROVED** [review]
LGTM
